### ISSUE TITLE
How to control what gcc compiler environment is associated with an spack Intel compiler?

### ISSUE THREAD

**Issue by garvct** [spack/spack#7626]
I have install an intel compiler intel@16.0.4 with spack and added it to the spack environment.

spack compiler list
==> Available compilers
-- gcc sles11-x86_64 --------------------------------------------
gcc@5.5.0  gcc@4.9.4  gcc@4.3.4

-- intel sles11-x86_64 ------------------------------------------
intel@16.0.4

How do I control what gcc compiler environment is associated with this Intel compiler (I do not want
to use the old system default, but a new gcc environment.)

e.g
spack env zlib%intel bash
which icc
/hpc-common/software/spack/0.11.2/spack/lib/spack/env/intel/icc
which gcc
/usr/bin/gcc

icc main.c
ldd a.out
	linux-vdso.so.1 =>  (0x00007fffedb06000)
	libm.so.6 => /lib64/libm.so.6 (0x00007fffed87c000)
	libgcc_s.so.1 => /lib64/libgcc_s.so.1 (0x00007fffed664000)
	libc.so.6 => /lib64/libc.so.6 (0x00007fffed2e8000)
	libdl.so.2 => /lib64/libdl.so.2 (0x00007fffed0e4000)
	/lib64/ld-linux-x86-64.so.2 (0x0000555555554000)

Outside spack, if I

module load gcc/5.5.0-gcc-4.3.4 intel/16.0.4-gcc-5.5.0

icc main.c
ldd a.out
linux-vdso.so.1 =>  (0x00007fffedb06000)
	libm.so.6 => /lib64/libm.so.6 (0x00007fffed87b000)
	libgcc_s.so.1 => /apps/local/spack/opt/linux-sles11-x86_64/gcc-4.3.4/gcc-5.5.0-upsc4w7bhfrlacvfu63xzvhfxw7l3y5f/lib64/libgcc_s.so.1 (0x00007fffed664000)
	libc.so.6 => /lib64/libc.so.6 (0x00007fffed2e8000)
	libdl.so.2 => /lib64/libdl.so.2 (0x00007fffed0e3000)
	/lib64/ld-linux-x86-64.so.2 (0x0000555555554000)

How do I associate an spack intel compiler with a newer gcc environment (i.e Not the system default gcc)?

Thanks,

**Comment by adamjstewart** [comment]
Does this help?
http://spack.readthedocs.io/en/latest/getting_started.html#intel-compilers

**Comment by garvct** [comment]
Yes, thanks this link helps, but it would be nice if spack could do this automatically by defining 
what gcc compiler is associated with an Intel compiler via the compilers.yaml file.

**Comment by adamjstewart** [comment]
You should be able to specify it in `cflags` and friends, but I agree that it is worth adding its own configuration variable.
